File: Random/4.Longest Consecutive Sequence.js
/*
    Author : Mohamed Yousef 
    Date   : 2022-12-16
*/
let solve=function(nums){
    let numSet=new Set(nums);
    longest=0
    console.log(numSet);
    numSet.forEach(ele => {
        if(numSet.has(ele-1)==false){
            length=0
            while(numSet.has(ele+length)==true){
                length++;
            }
            longest=Math.max(length,longest)
        }
    })
    return longest
}
console.log(solve([-1,-9,-5,-2,-9,8,-8,1,-9,-3,-3]));

File: Random/4.Longest Consecutive Sequence.py
#    Author : Mohamed Yousef 
#    Date   : 2022-12-15
import sys,math,bisect,collections,itertools,heapq
from collections import defaultdict,deque
sys.setrecursionlimit(50000) #To increase Recursion depth in py
def solve(nums):
    numSet=set(nums)
    longest=0
    for n in numSet:
        if n-1 not in numSet:
            length=0
            while n+length in numSet:
                length+=1
            longest=max(longest,length)
    return longest

print(solve([0,3,7,2,5,8,4,6,0,1]))